examples/static-contact-dynamics.py

```diff
@@ -0,0 +1,166 @@
+import numpy as np
+import pinocchio as pin
+
+from os.path import dirname, join, abspath
+np.set_printoptions(linewidth=np.inf)
+
+# ----- PROBLEM STATEMENT ------
+#
+# We want to find the contact forces and torques required to stand still at a configuration 'q0'.
+# We assume 3D contacts at each of the feet
+#
+# The dynamic equation would look like:
+#
+# M*q_ddot + g(q) + C(q, q_dot) = tau + J^T*lambda --> (for the static case) --> g(q) = tau + Jc^T*lambda (1)
+
+# ----- SOLVING STRATEGY ------
+
+# Split the equation between the base link (_bl) joint and the rest of the joints (_j). That is,
+#
+#  | g_bl |   |  0  |   | Jc__feet_bl.T |   | l1 |
+#  | g_j  | = | tau | + | Jc__feet_j.T  | * | l2 |    (2)
+#                                           | l3 |
+#                                           | l4 |
+
+# First, find the contact forces l1, l2, l3, l4 (these are 3 dimensional) by solving for the first 6 rows of (2).
+# That is,
+#
+# g_bl   = Jc__feet_bl.T * | l1 |
+#                          | l2 |    (3)
+#                          | l3 |
+#                          | l4 |
+#
+# Thus we find the contact froces by computing the jacobian pseudoinverse,
+#
+# | l1 | = pinv(Jc__feet_bl.T) * g_bl  (4)
+# | l2 |
+# | l3 |
+# | l4 |
+#
+# Now, we can find the necessary torques using the bottom rows in (2). That is,
+#
+#                             | l1 |
+#  tau = g_j - Jc__feet_j.T * | l2 |    (5)
+#                             | l3 |
+#                             | l4 |
+
+# ----- SOLUTION ------
+
+# 0. DATA
+pinocchio_model_dir = join(dirname(dirname(str(abspath(__file__)))), "models")
+
+model_path = join(pinocchio_model_dir, "example-robot-data/robots")
+mesh_dir = pinocchio_model_dir
+urdf_filename = "solo12.urdf"
```

Comment:
I would use a humanoid robot, so we can combine 3d and 6d contacts.

Reply:
Solo is very common as a template example. No reason to change.